Feature/vmp 54 integration test framework

Kamyki opened the pull request:
Beep beep, here it is.

w pre-integration-test: buduje graf dla bydoszczy z danych wpakowanych do `rosources/graph/bydgosz`
w integration-test: failsafe odpala testy integracyjne które wczytują wbudowany graf

Rzeczy które bym zmienił:
- aplikacja wstaje przez wybebeszenie niektórych rzeczy z maina. W następnym refaktorze zmieniłbym maina by wszstko było tam statyczne i każda medota przyjmowałą OTPServer itp. Pozwoliłoby to użyć gotowych metodod.
- aplikacja wstaje dla każdego testu oddzielnie
- nie jestem pewien czy test czeka aż zaczyta parkingZones i vehicles gdzyż one działają w innym wątku, podejrzewam, że nie

Kamyki committed:
add integration test configuration

Kamyki committed:
refactor OTPMain

Kamyki committed:
add example integration tests

Kamyki committed:
Fix jackson serialization issues

Kamyki committed:
move resources to integration test

AdamWiktor commented on src/integrationTest/java/org/opentripplanner/IntegrationTest.java:
```diff
@@ -0,0 +1,59 @@
+    @Override
+    public Application configure() {
+        set(TestProperties.CONTAINER_PORT, port);
+        System.setProperty("sharedVehiclesApi", "http://ns3114244.ip-54-38-192.eu:8888/get_standard_vehicle_positions"); // mocked database
```
To jest publiczne repo, nie wiem czy to dobry pomysł wstawiać tutaj nasz własny URL. Może np wstawić ten url jako argument wywołania testów w wierszu poleceń?

Kamyki replied:
można, ale docelowo fajnie by było postawić wiremocka czy coś

Kamyki replied:
Bez tego odpalanie pojedyńczych tesów z IDE będzie raczej karkołomne

AdamWiktor replied:
to prawda, wiremock byłby idealny

AdamWiktor commented on src/integrationTest/java/org/opentripplanner/base/ResponseStructureIT.java:
```diff
@@ -0,0 +1,52 @@
+        Response body = response.readEntity(Response.class);
+
+        assertEquals("should return status 200", 200, response.getStatus());
+        assertEquals("should return 3 tracks", 1, body.getPlan().itinerary.size());
```
Coś tu się nie zgadza xD

Kamyki replied:
copy paste :D

AdamWiktor commented on src/main/java/org/opentripplanner/standalone/OTPMain.java:
```diff
@@ -43,7 +43,15 @@
     public static void main(String[] args) {
 
-        /* Parse and validate command line parameters. */
+        CommandLineParameters params = parseCommandLineParams(args);
```
Bardzo dobre zmiany w tym pliku 👍

AdamWiktor commented:
Nie wiem, czy możemy tak sobie wstawić plik OSM i GTFS do publicznego repo - kwestie licencji i takie tam. Może dałoby się to ściągać z podanych URLi przed uruchomieniem testów?

Kamyki commented:
Można tylko pamiętaj, że w takim razie będzie to ściągane w każdym buildzie na githubie

AdamWiktor commented:
> Można tylko pamiętaj, że w takim razie będzie to ściągane w każdym buildzie na githubie

Wiem, ale mimo wszystko trochę mnie martwi wrzucanie takich plików do repo - raz, że są duże, a dwa że to jednak jest hardkodowanie całej mapy Bydgoszczy i rozkładów jazdy dla jakiegoś konkretnego momentu w czasie

MironSz commented:
OSMy na pewno mają licencję która zezwala na wstawienie tego (wraz z licencją ofc), nie wiem jak z Bydgoszczem

Kamyki committed:
Refactor integration tests

- make main class static
- initialise graph only once per test class
- add wiremock to vehicles and parking zones
- configure timeouts to 10s to make runs more consistent

Kamyki committed:
Fix gitignore

Kamyki committed:
Code review changes
